src/lib/operators/table_scan/column_between_table_scan_impl.cpp

```diff
@@ -54,19 +56,34 @@ void ColumnBetweenTableScanImpl::_scan_generic_segment(const BaseSegment& segmen
 
     auto typed_left_value = type_cast_variant<ColumnDataType>(_left_value);
     auto typed_right_value = type_cast_variant<ColumnDataType>(_right_value);
-    auto comparator = [typed_left_value, typed_right_value](const auto& position) {
-      return position.value() >= typed_left_value && position.value() <= typed_right_value;
-    };
 
-    _scan_with_iterators<true>(comparator, it, end, chunk_id, matches);
+    with_comparator_between(_predicate_condition, [&](auto lower_comparator, auto upper_comparator) {
+      auto between_comparator = [&](const auto& position) {
+        return lower_comparator(position.value(), typed_left_value) &&
+               upper_comparator(position.value(), typed_right_value);
+      };
+      _scan_with_iterators<true>(between_comparator, it, end, chunk_id, matches);
+    });
   });
 }
 
 void ColumnBetweenTableScanImpl::_scan_dictionary_segment(const BaseDictionarySegment& segment, const ChunkID chunk_id,
                                                           PosList& matches,
                                                           const std::shared_ptr<const PosList>& position_filter) const {
-  auto left_value_id = segment.lower_bound(_left_value);
-  auto right_value_id = segment.upper_bound(_right_value);
+  // naming assumption: the left value is always the lower one (otherwise the result is empty)
```

Comment:
where is this enforced?

Reply:
nowhere, because it is valid input. This comment only struggles with the confusing naming of left_values and lower_values...
My problem with this naming is, that the left_value is always the left_value, but the lower_value is not necessarily the lower one.

My confusion is e.g. line 75-76 where the check is for lower-inclusiveness and then takes the left_value...

Solution? For me, the comment seemed to be the easiest solution to put attention on this

Reply:
I think there is no reason for confusion here. If lower_bound > upper_bound than that's just the case and the result of the Between is empty. Still, lower bound is the lower bound and upper bound is the upper bound. They just do not happen to produce a non-empty intersection :man_shrugging: